program/rust/src/rust_oracle.rs

```diff
@@ -150,3 +183,38 @@ fn load_account_as_mut<'a, T: Pod>(
         bytemuck::from_bytes_mut(&mut data[0..size_of::<T>()])
     }))
 }
+
+/// Mutably borrow the data in `account` as a mapping account, validating that the account
+/// is properly formatted. Any mutations to the returned value will be reflected in the
+/// account data. Use this to read already-initialized accounts.
+fn load_mapping_account_mut<'a>(
+    account: &'a AccountInfo,
+    expected_version: u32,
+) -> Result<RefMut<'a, pc_map_table_t>, ProgramError> {
+    let mapping_account_ref = load_account_as_mut::<pc_map_table_t>(account)?;
+    let mapping_account = *mapping_account_ref;
+
+    pyth_assert(
+        mapping_account.magic_ == PC_MAGIC
+            && mapping_account.ver_ == expected_version
+            && mapping_account.type_ == PC_ACCTYPE_MAPPING,
+        ProgramError::InvalidArgument,
+    )?;
+
+    Ok(mapping_account_ref)
+}
+
+/// Initialize account as a new mapping account. This function will zero out any existing data in
+/// the account.
+fn initialize_mapping_account(account: &AccountInfo, version: u32) -> Result<(), ProgramError> {
+    clear_account(account)?;
```

Comment:
I wonder if it's a good idea to only keep instructions in oracle.rs, and move helpers into other modules. (It currently looks very clean, am just imagining the file will grow a lot as we migrate more instructions)

Reply:
This is bound to happen at some time. Arguably each instruction could have its own file also.

Reply:
we can move stuff around later. i agree this file is going to get too big, but it's not yet clear to me what the right split into multiple files is.